src/storage.rs

```diff
@@ -12,6 +46,115 @@ impl Default for FileStorage {
     }
 }
 
-pub trait Storage {}
+pub trait Storage {
+    fn get_username(&self) -> String;
+    fn get_password(&self) -> String;
+    fn get_token(&self) -> String;
```

Comment:
For reading values, it is generally better to use references, so the value doesn't need to be copied each time:

```rust
fn get_token(&self) -> &str;
// ...
fn get_token(&self) -> &str {
    &self.token
}